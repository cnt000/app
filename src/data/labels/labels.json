{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
