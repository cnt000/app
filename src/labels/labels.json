{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
