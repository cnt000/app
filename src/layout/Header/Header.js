import React from 'react';
import logo from '../../images/logo.png';
import styles from './Header.module.css';

const Header = ({ openMenuLabel, title, titleDescription, goToCartLabel, setMenuOpen }) => {
  return (
    <header className={styles.header}>
      <div>
        <button className={styles.burger} onClick={setMenuOpen}>{openMenuLabel}</button>
      </div>
      <div className={styles.content}>
        <div className={styles.logo}>
          <img src={logo} alt="pungilandia logo" />
        </div>
        <div className={styles.title}>
          <h1>{title}</h1>
          <p>{titleDescription}</p>
        </div>
      </div>
      <div>
        <button className={styles.cart}>{goToCartLabel}</button>
      </div>
    </header>
  );};

export default Header;

