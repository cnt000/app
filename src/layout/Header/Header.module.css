.header {
  grid-area: header;
  display: grid;
  grid-template-columns: 2.6rem 1fr 3rem;
}

.burger {
  webkit-appearance: button;
  -moz-appearance: button;
  appearance: button;
  background-color: transparent;
  text-decoration: none;
  color: initial;
  cursor: pointer;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
  padding: 0;
  border: 0;
  background-image: url('../../images/menu.svg');
  background-repeat: no-repeat;
  background-size: 2.3rem 3.3rem;
  margin-left: 0.4rem;
  height: 100%;
}

.logo {

}

.content {
  text-align: left;
  display: flex;
  margin: 0 auto;
  color: var(--brown);
}

.content h1 {
  font-size: 32px;
  text-shadow: 2px 2px 16px var(--grey);
}

.content p {
  font-size: 16px;
  text-shadow: 2px 2px 16px var(--grey);
}

.title {
  padding: 0 0.4rem;
  font-family: 'Caveat Brush', sans-serif;
  line-height: 0.9;
}

.logo img {
  width: auto;
  height: 42px;
}

.cart {
  appearance: button;
  background-color: transparent;
  text-decoration: none;
  color: initial;
  cursor: pointer;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
  padding: 0;
  border: 0;
  background-image: url('../../images/cart.svg');
  background-repeat: no-repeat;
  text-indent: -9999px;
  width: 3rem;
  height: 2.4rem;
  background-size: 2.8rem 2.2rem;
  margin-top: 0.5rem;
  margin-right: 0.4rem;
  height: 100%;
}

@media only screen and (min-width: 1025px) {
  .header {
    grid-template-columns: 3rem 1fr 3rem;
  }

  .burger {
    display: none;
  }

  .content h1 {
    font-size: 86px;
    text-shadow: 2px 2px 16px var(--grey);
  }

  .content p {
    font-size: 42px;
    text-shadow: 2px 2px 16px var(--grey);
  }

  .cercapiante {
    position: absolute;
    right: 1rem;
    line-height: 3.4;
  }

  .logo img {
    height: 115px;
  }
}
