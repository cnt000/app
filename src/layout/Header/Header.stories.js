import React from 'react';
import { action } from '@storybook/addon-actions';

import Header from './Header';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/labels';

export default {
  component: Header,
  title: 'Header',
};

const actionsData = {
  setMenuOpen: action('setMenuOpen(true)'),
};

export const Default = () => (
  <LabelsContext.Provider value={labels}>
    <Header {...actionsData} />
  </LabelsContext.Provider>
);
