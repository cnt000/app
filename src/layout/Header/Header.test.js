import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';

import Header from './Header';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/Labels';

const setMenuOpenMock = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/example/path',
  }),
}));

const header = (
  <Router>
    <LabelsContext.Provider value={labels}>
      <Header setMenuOpen={() => setMenuOpenMock(true)} />
    </LabelsContext.Provider>
  </Router>
);

it('renders correctly', () => {
  const tree = renderer.create(header).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Logo title and subtitle', () => {
  const { getByText } = render(header);
  expect(getByText(/Pungilandia/)).toBeInTheDocument();
  expect(getByText(/Piante grasse, strane ed insolite/)).toBeInTheDocument();
});

describe('Click open menu', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('click open button call setMenuOpenMock', () => {
    act(() => {
      ReactDOM.render(header, container);
    });
    const button = container.querySelector('button.burger');

    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(setMenuOpenMock).toHaveBeenCalled();
  });
});
