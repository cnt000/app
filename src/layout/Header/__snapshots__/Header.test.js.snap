// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <div>
    <button
      className="burger"
      onClick={[Function]}
    >
      apri il menu
    </button>
  </div>
  <div
    className="content"
  >
    <div
      className="logo"
    >
      <img
        alt="pungilandia logo"
        src="logo.png"
      />
    </div>
    <div
      className="title"
    >
      <h1>
        Pungilandia
      </h1>
      <p>
        Piante grasse, strane ed insolite
      </p>
    </div>
  </div>
  <div>
    <button
      className="cart"
    >
      vai al carrello
    </button>
  </div>
</header>
`;
